name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/ComdirectPDFExtractor.java

```diff
@@ -1155,170 +1728,332 @@ else if (v.get("note1").matches("^(?i:Visa\\-Kartenabre.*)$"))
+                        // is greater than the calculated gross amount (withholding tax).
+                        // @formatter:on
+                        if (taxesTransaction.getSecurity().getAttributes()
+                                        .get(new AttributeType(ATTRIBUTE_GROSS_TAXES_TREATMENT)) != null
+                                        && ((Money) taxesTransaction.getSecurity().getAttributes() //
+                                                        .get(new AttributeType(ATTRIBUTE_GROSS_TAXES_TREATMENT)))
+                                                                        .isGreaterThan(dividendTransaction
+                                                                                        .getGrossValue()))
+                        {
+                            dividendTransaction.setMonetaryAmount((Money) taxesTransaction.getSecurity().getAttributes()
+                                            .get(new AttributeType(ATTRIBUTE_GROSS_TAXES_TREATMENT)));
+
+                            ExtractorUtils.fixGrossValue().accept(dividendTransaction);
+                        }
+
+                        // Subtract the taxes from the taxes transaction from the total amount
+                        dividendTransaction.setMonetaryAmount(dividendTransaction.getMonetaryAmount()
+                                        .subtract(taxesTransaction.getMonetaryAmount()));
+
+                        // Add taxes as tax unit
+                        dividendTransaction.addUnit(new Unit(Unit.Type.TAX, taxesTransaction.getMonetaryAmount()));
+
+                        // Combine at sources file (One or two files?)
+                        if (!dividendTransaction.getSource().equals(taxesTransaction.getSource()))
+                            dividendTransaction.setSource(dividendTransaction.getSource() + "; " + taxesTransaction.getSource());
+
+                        // Combine at notes
+                        dividendTransaction.setNote(concat(dividendTransaction.getNote(), taxesTransaction.getNote()));
+
+                        // Set note that the taxes transaction will be deleted
+                        taxesTransaction.setNote(ATTRIBUTE_TO_BE_DELETED);
+                    }
+                    else
+                    {
+                        // do nothing because no taxes transaction is present
+                    }
+                }
+            });
+        });
+
+        // iterate list and remove items that are marked ATTRIBUTE_TO_BE_DELETED
+        Iterator<Item> iter = items.iterator();
+        while (iter.hasNext())
         {
-            Annotated s = txi.getSubject();
-            if (s instanceof AccountTransaction a)
+            Object o = iter.next().getSubject();
+            if (o instanceof AccountTransaction a)
             {
-                return AccountTransaction.Type.DIVIDENDS.equals(a.getType());
+                if (ATTRIBUTE_TO_BE_DELETED.equals(a.getNote()))
```

Comment:
While the deletion logic is not wrong, I wonder if we can make it easier by using a Set:

```java
Set<AccountTransaction> toBeDeleted = new HashSet<>();

...

toBeDeleted.add(taxesTransaction)

...

Iterator<Item> iter = items.iterator();
while (iter.hasNext())
{
    Object o = iter.next().getSubject();
    if (toBeDeleted.contains(o))
        iter.remove();
}
```

(or, if you kept the item around, then ```items.remove(taxesTransactionItem)```)